src/gsi/m_berror_stats_reg.f90

```diff
@@ -290,6 +326,817 @@ subroutine berror_read_bal_reg(msig,mlat,agvi,bvi,wgvi,mype,unit)
+     corz(1:mlat,1:nsig,:,nrf3_sfwter)=corz(1:mlat,1:nsig,:,nrf3_sf)
+     hwll(0:mlat+1,1:nsig,:,nrf3_sfwter)=hwll(0:mlat+1,1:nsig,:,nrf3_sf)
+  endif
+
+  if (nrf3_vpwter>0) then
+     if(mype==0) write(6,*)'Replace default with appropriate statistics for variable vpwter'
+     corz(1:mlat,1:nsig,:,nrf3_vpwter)=corz(1:mlat,1:nsig,:,nrf3_vp)
+     hwll(0:mlat+1,1:nsig,:,nrf3_vpwter)=hwll(0:mlat+1,1:nsig,:,nrf3_vp)
+  endif
+
+! 2d variable
+  do n=1,nc2d
+     loc=nrf2_loc(n)
+     if (nrf_err(loc)) cycle
+     if (n==nrf2_sst) then
+        do i=1,mlat
+           corp(i,:,n)=zero_3
+        end do
+        do i=0,mlat+1
+           if( if_cs_staticB )then ! When convective-scale B is used, sf is not available
+              hwllp(i,:,n)=hwll(i,1,:,nrf3_t)
+              hwllp(i,:,nc2d+1)=hwll(i,1,:,nrf3_t)
+              hwllp(i,:,nc2d+2)=hwll(i,1,:,nrf3_t)
+           else
+              hwllp(i,:,n)=hwll(i,1,:,nrf3_sf)
+              hwllp(i,:,nc2d+1)=hwll(i,1,:,nrf3_sf) !not very nice, since it assumes that stl and sti
+              hwllp(i,:,nc2d+2)=hwll(i,1,:,nrf3_sf) !are always the first motley variables in convinfo
+           end if
+        end do
+     end if
+     if (n==nrf2_gust) then
+        do i=1,mlat
+           corp(i,:,n)=three
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwll(i,1,:,nrf3_q)
+        end do
+     end if
+     if (n==nrf2_vis) then
+        do i=1,mlat
+           corp(i,:,n)=20000.0_r_kind
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwll(i,1,:,nrf3_t)
+        end do
+     end if
+     if (n==nrf2_pblh) then
+        do i=1,mlat
+           corp(i,:,n)=500.0_r_kind
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=three*hwll(i,1,:,nrf3_t)
+        end do
+     end if
+     if (n==nrf2_wspd10m) then
+        do i=1,mlat
+           corp(i,:,n)=three
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwll(i,1,:,nrf3_q)
+        end do
+     end if
+     if (n==nrf2_td2m) then
+        raux=maxval(corz(1:mlat,1,:,nrf3_q))
+        do i=1,mlat
+           corp(i,:,n)=(corz(i,1,:,nrf3_q)/raux)*three !tentatively
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwll(i,1,:,nrf3_q) !tentatively
+        end do
+     end if
+     if (n==nrf2_mxtm) then
+        do i=1,mlat
+           corp(i,:,n)=corz(i,1,:,nrf3_t)
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwll(i,1,:,nrf3_t)
+        end do
+     end if
+     if (n==nrf2_mitm) then
+        do i=1,mlat
+           corp(i,:,n)=corz(i,1,:,nrf3_t)
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwll(i,1,:,nrf3_t)
+        end do
+     end if
+     if (n==nrf2_pmsl) then
+        do i=1,mlat
+           corp(i,:,n)=corp(i,:,nrf2_ps)
+        end do
+        do i=0,mlat+1
+           hwllp(i,:,n)=hwllp(i,:,nrf2_ps)
+        end do
+     end if
+     if (n==nrf2_howv) then
+        call read_howv_stats(mlat,1,2,cov_dum)
```

Comment:
It should be changed to read_howv_stats(mlat,1,2,cov_dum,**mype**) in the latest code.

Reply:
Corrected.